staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -2246,7 +2246,7 @@ func TestGetWatchCacheResourceVersion(t *testing.T) {
 				require.NoError(t, err)
 			}
 
-			actualResourceVersion, err := cacher.getWatchCacheResourceVersion(context.TODO(), uint64(parsedResourceVersion), scenario.opts)
+			actualResourceVersion, err := cacher.getWatchCacheResourceVersion(context.TODO(), uint64(parsedResourceVersion), scenario.opts, true)
```

Comment:
The `TestGetWatchCacheResourceVersion` tests all combinations of `ResourceVersion`, `AllowWatchBookmarks` and `SendInitialEvents`.

Once we are okay with how `getWatchCacheResourceVersion` works we could extend this test by `isRequestWatchProgressSupported`.